### PR TITLE
Chat prompt history

### PR DESCRIPTION
**Pull request by InAnYan**
Closes https://github.com/InAnYan/jabref/issues/149.

This PR gives users the ability to scroll past user messages in AI chat.

Similar to shells command history.

The algorithm looks clunky, but I tested it, and it seems to work good.

### Mandatory checks

~- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)~
~- [ ] Tests created for changes (if applicable)~
- [x] Manually tested changed features in running JabRef (always required)
~- [ ] Screenshots added in PR description (for UI changes)~
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/gui/ai/components/aichat/AiChatComponent.java [hunk 1]
```diff
@@ -41,6 +45,10 @@ public class AiChatComponent extends VBox {
     private final TaskExecutor taskExecutor;
 
     private final IntegerProperty blockScroll = new SimpleIntegerProperty(0);
+    // -1 for empty, new, or non-existent messages.
+    // Is user scrolls for messages, then it will be non-negative.
+    private final IntegerProperty currentUserMessageScroll = new SimpleIntegerProperty(-1);
```
Comment (koppor) [inline src/main/java/org/jabref/gui/ai/components/aichat/AiChatComponent.java:50]: Introduce a constant for `-1`: `NEW_EMPTY_NON_EXISTENT`.

Code should not contain magic numbers. With a reference to the constant ( `private static final`), it is possible to trace the constant.

This also gets rid of comment line 48.
line 49 can be kept

File: src/main/java/org/jabref/gui/ai/components/aichat/AiChatComponent.java [hunk 2]
```diff
@@ -65,11 +73,44 @@ public AiChatComponent(AiPreferences aiPreferences, AiChatLogic aiChatLogic, Str
...
+                    showingHistoryMessage.set(true);
+                    currentUserMessageScroll.set(currentUserMessageScroll.get() - 1);
+                }
+            } else if (keyEvent.getCode() == KeyCode.UP) {
+                if ((currentUserMessageScroll.get() < getReversedUserMessagesStream().count() - 1) && (userPromptTextArea.getText().isEmpty() || showingHistoryMessage.get())) {
+                    showingHistoryMessage.set(true);
+                    currentUserMessageScroll.set(currentUserMessageScroll.get() + 1);
+                }
+            } else {
+                if (keyEvent.getCode() != KeyCode.RIGHT && keyEvent.getCode() != KeyCode.LEFT) {
```
Comment (koppor) [inline src/main/java/org/jabref/gui/ai/components/aichat/AiChatComponent.java:87]: Add comment abvoe

    // Cursor left/right should not stop history scrolling

**Review by koppor — CHANGES_REQUESTED** [review]

**Comment by InAnYan** [comment]
I wanted to make a separate component `HistoryTextField`, but it didn't go well.

I've added the constant and explanation in comments about how the history is handled

**Comment by koppor** [comment]
> I wanted to make a separate component `HistoryTextField`, but it didn't go well.

Great idea!

As usual: Create a new minimal JavaFX project. Show current progress there.

My hope: As team, we can get it done - and then submit a PR to https://github.com/dlsc-software-consulting-gmbh/GemsFX.

File: src/main/java/org/jabref/gui/ai/components/aichat/AiChatComponent.java [hunk 3]
```diff
@@ -42,6 +50,15 @@ public class AiChatComponent extends VBox {
 
     private final IntegerProperty blockScroll = new SimpleIntegerProperty(0);
 
+    // This property stores index of a user history message.
+    // When user scrolls history in the prompt, this value is updated.
+    // Whenever user edits the prompt, this value is reset to NEW_NON_EXISTENT_MESSAGE.
+    private final IntegerProperty currentUserMessageScroll = new SimpleIntegerProperty(-1);
```
Comment (koppor) [inline src/main/java/org/jabref/gui/ai/components/aichat/AiChatComponent.java:56]:
```suggestion
    private final IntegerProperty currentUserMessageScroll = new SimpleIntegerProperty(NEW_NON_EXISTENT_MESSAGE);
```

**Review by koppor — CHANGES_REQUESTED** [review]
Constants should be used everywhere :) - not just in 90% of the cases 😅🤣🤣

**Review by koppor — APPROVED** [review]

**Pull request merged**